### DISCUSSION TITLE
Table of itwin-iui slow

### DISCUSSION THREAD

**Discussion by RaphaelVanierBentley** [iTwin/iTwinUI#1049] [Q&A]
Hello

I have a problem right now. I am trying to use the **<Table ... />** but it's pretty slow. In my use case I have about 920 rows. I have reproduced the case in a Sandbox:

[](https://www.itwinjs.org/sandbox/RaphaelVanier/itwin-iui-Table-lag)

It takes about 3s to generate. When debugging locally it takes 10s+. If you expect lag from the backend to generate the results, you will need the most speed at all levels. Here is the same less beautiful example in pure HTML:

[](https://www.itwinjs.org/sandbox/RaphaelVanier/itwin-iui-Table-pure-html/)

The generation is almost instantaneous.

Is there any way to optimize the <Table ... /> tool to display the results in less than 1s?

Thanks

**Comment by mayank99** [comment]
hi @RaphaelVanierBentley, i think there are some links missing in your post. can you double check?

also, have you tried using the `enableVirtualization` prop with Table? https://itwin.github.io/iTwinUI/react/?path=/story/core-table--virtualized
